doc/sphinx/source/esmvalcore/preprocessor.inc

```diff
@@ -1,36 +1,215 @@
+masked out, respectively. If the corresponding fx file is not found (which is
+the case for some models and almost all observational datasets), the
+preprocessor attempts to mask the data using Natural Earth mask files (that are
+vectorized rasters). As mentioned above, the spatial resolution of the the
+Natural Earth masks are much higher than any typical global model (10m for
+land and 50m for ocean masks).
+
+Note that for masking out ice sheets, the preprocessor uses a different
+function, to ensure that both land and sea or ice can be masked out without
+losing generality. To mask ice out, `mask_landseaice` can be used:
+
+.. code-block:: yaml
+
+  preprocessors:
+    preproc_mask:
+      mask_landseaice:
+        mask_out: ice
+
+and requires only one argument:
+* mask_out: either landsea or ice.
+
+As in the case of `mask_landsea`, the preprocessor automatically retrieves the
+`fx: sftgif` mask.
+
+Another option is to just read the fx masks as any other CMOR variable and use
+it within a diagnostic script. This can be done in the variable dictionary by
+specifiying the desired fx variables (masks):
+
+.. warning::
+  Code snippet, text and link to function to be added (after #1037 and #1075
+  are closed).
+
+
+
+Horizontal regridding
+=====================
+.. warning::
+   Documentation of  _regrid.py (part 2) to be added
+
+
+
+Masking of missing values
+=========================
+.. warning::
+   Documentation of  _mask.py (part 2) to be added
+
+
+Multi-model statistics
+======================
+
+.. warning::
+   Documentation of _multimodel.py to be added.
+
+Information on maximum memory required: In the most general case, we can set upper limits on the maximum memory the anlysis will require:
+
+Ms = (R + N) x F_eff - F_eff - when no multimodel analysis is performed;
+Mm = (2R + N) x F_eff - 2F_eff - when multimodel analysis is performed;
+
+where
+
+Ms: maximum memory for non-multimodel module
+Mm: maximum memory for multimodel module
+R: computational efficiency of module (typically 2-3)
+N: number of datasets
+F_eff: average size of data per dataset where F_eff = e x f x F
+where e is the factor that describes how lazy the data is (e = 1 for fully realized data)
+and f describes how much the data was shrunk by the immediately previous module eg
+time extraction, area selection or level extraction; note that for fix_data f relates only to the time extraction, if data is exact in time (no time selection) f = 1 for fix_data
+
+so for cases when we deal with a lot of datasets (R + N = N), data is fully realized, assuming an average size of 1.5GB for 10 years of 3D netCDF data, N datasets will require
+
+
+Ms = 1.5 x (N - 1) GB
+Mm = 1.5 x (N - 2) GB
+
+
+Time operations
+===============
+
+The _time.py module contains a broad set of functions to subset data and apply
+statistical operators along the temporal coordinate of the input data:
+
+1. extract_time: extract a time range from a cube.
```

Comment:
Does this mean it extracts a period that is shorter than the original data set? Or does this mean that the time period of the data set is extracted?

Reply:
A perioed that is shorter than the original one.

Reply:
Maybe add then "specified" so that it reads "extract a specified time range from the cube"

Reply:
Done.